### DISCUSSION TITLE
is there a way to get vCenter Events in the last 5 minutes ?

### DISCUSSION THREAD

**Discussion by manjunath-batakurki** [vmware/govmomi#2942] [Community Help] [answered]
Hi team,

I'm able to get the events from the vCenter using the Event Manage(github.com/vmware/govmomi/event)r. I see that it gives the events based on the PageSize specified to Events() function. is there a way to get all the vCenter Events that happened in a given interval say last 5 minutes ?

Thank you very much in advance :)

**Comment by embano1** [comment, marked as answer]
Yes, see this example: https://github.com/vmware/govmomi/blob/master/examples/events/main.go#L51-L59

You can also use this convenience library I built: https://pkg.go.dev/github.com/embano1/vsphere@v0.2.4/event#WithTime

**Reply by manjunath-batakurki** [reply]
Thank you. This helps a lot :) . I was able to filter events based on time. one disadvantage of using `EventFilterSpec` is that i cannot filter based on multiple events like `VmPoweredOffEvent` and `VmPoweredOnEvent` in at one time as **EventTypeId** works with only one event..

**Reply by embano1** [reply]
Welcome!

> one disadvantage of using EventFilterSpec is that i cannot filter based on multiple events like VmPoweredOffEvent and VmPoweredOnEvent in at one time as EventTypeId works with only one event..

IMHO this is not correct. Are you using `govmomi` or my library? Both of them support `EventTypeId` to be of [`[]string`](https://vdc-repo.vmware.com/vmwb-repository/dcr-public/6b586ed2-655c-49d9-9029-bc416323cb22/fa0b429a-a695-4c11-b7d2-2cbc284049dc/doc/vim.event.EventFilterSpec.html). Did you try passing a slice of event type strings?

**Reply by manjunath-batakurki** [reply]
@embano1 Sorry. It works as expected. it does take multiple events as input. Thank you so much for all the help...
